backend/src/card_game/api/base.clj

```diff
@@ -0,0 +1,40 @@
+(ns card-game.api.base
+  (:require [card-game.core :as core]
+            [card-game.victory-conditions :as victory-conditions]
+            [card-game.persistence :as persistence]
+            [card-game.api.helper :as helper]
+            [configs :as configs]))
+
+(defn get-game
+  "Fetches a game from an ID and returns the visible part as a player"
+  [game-id player-id]
+  (let [game-state (persistence/fetch-game game-id)]
+  (assoc (helper/get-game-as-player game-state player-id)
+         :status (helper/define-status game-state player-id))))
+
+(defn play-card-as-player
+  [game-id player index row]
+  (let [game-state (persistence/fetch-game game-id)]
+      (if (= (:status (get-game game-id player)) configs/play)
+          (do
+            (persistence/save-game (core/play-card game-state (helper/player-num game-state player) index row))
+            (get-game game-id player))
+          {:error configs/out-of-turn})))
+
+(defn add-player
+  "Adds a player to a game"
+  [game-id]
+  (let [saved-game (persistence/fetch-game game-id)
+        players-connected (count (:player-ids saved-game))]
+    (if (> players-connected 1)
```

Comment:
I just realized a way to simplify this function would be to move the check for "too many players" onto `helper/create-player` (or am I missing something?)

Reply:
Right now `add-player` returns `{:error configs/too-players}` when too many players.
Making the check on `helper/create-players` (without many other changes) would imply calling `(get-game` with an `{:error configs/too-players}` game-state. I think the result of that might be a cause of bugs.

Reply:
🤔 maybe... I don't know, `add-player` is one function I'm not very happy about, I just feel like it does too many things:

1. Fetches a game
2. Checks how many players the game has and returns an error if there are too many
3. If the game doesn't exist, it creates it.
4. It adds a player to the game.
5. It fetches the game as a player and returns it. <-- This I will make an issue to change accross the board so ignore it

I feel like 2 and 3 shouldn't be the responsibility of a function called `add-player`

Reply:
I feel like `add-players` should check if you can actually add that player, so I don't see a problem with 2.
Agree with 3.

Reply:
Ok, I can agree to that.